Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -968,28 +990,28 @@ void DirectoryBlobPartition::SetDiscType(std::optional<bool> is_wii)
 
 void DirectoryBlobPartition::SetBI2FromFile(const std::string& bi2_path)
 {
-  m_bi2.resize(BI2_SIZE);
+  std::vector<u8> bi2(BI2_SIZE);
 
   if (!m_is_wii)
-    Write32(INVALID_REGION, 0x18, &m_bi2);
+    Write32(INVALID_REGION, 0x18, &bi2);
 
-  const size_t bytes_read = ReadFileToVector(bi2_path, &m_bi2);
+  const size_t bytes_read = ReadFileToVector(bi2_path, &bi2);
   if (!m_is_wii && bytes_read < 0x1C)
     ERROR_LOG_FMT(DISCIO, "Couldn't read region from {}", bi2_path);
 
-  m_contents.AddReference(BI2_ADDRESS, m_bi2);
+  m_contents.Add(BI2_ADDRESS, bi2);
 }
 
 void DirectoryBlobPartition::SetBI2(std::vector<u8> bi2)
 {
   const size_t bi2_size = bi2.size();
-  m_bi2 = std::move(bi2);
-  m_bi2.resize(BI2_SIZE);
+  std::vector<u8> new_bi2 = bi2;
```

Comment:
No need to copy the vector, since `bi2` isn't a reference. Just use `bi2` directly instead.